### ISSUE TITLE
steamwebhelper keeps crashing on startup

### ISSUE THREAD

**Issue by VojislavMrdja** [ValveSoftware/steam-for-linux#11262]
#### Your system information
[fastfetch.txt](https://github.com/user-attachments/files/17006547/fastfetch.txt)
* Steam client version (build number or date): steam 1.0.0.81-2 from the arch repo
* Distribution (e.g. Ubuntu): Arch (Linux 6.10.10-arch1-1)
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes
* Steam Logs:  [logs.tar.gz](https://github.com/user-attachments/files/17006548/logs.tar.gz)
* GPU: Radeon R7 M340 [Discrete] & Intel HD Graphics 620 [Integrated]
* Today's pacman log for good measure:  [pacmanlog.txt](https://github.com/user-attachments/files/17006563/pacmanlog.txt)

#### Please describe your issue in as much detail as possible:

Steam used to work fine until i ran a pacman -Syu today (09/15/2024) and steam just crashes on startup now, i don't have any weird symlinks or anything like that, just a standard steam install from the main arch repo. i tried running it with -vgui no change, with -cef-dissable-sanbox and -cef-disable-gpu also no change, i sometimes get the steamwebhelper has crashed what do you want to do window, but only sometimes, and other times it just keeps silently crashing and looping the log message in the terminal, closest issues i could find are #10548 , #10547 and #10560 but again i don't have any symlinks in place

#### Steps for reproducing this issue:

1. Open Steam via the desktop file or via the terminal
2. steamwebhelper keeps crashing and steam just keeps loading forever
3. no profit...

**Comment by VojislavMrdja** [comment]
All right, so I left the laptop open for a while and I just tried opening steam and it opened without issue, what was the problem? no idea, is it going to reoccur? again no idea, what fixed it? biggest mystery since the release date of portal 3. 
I really hate those kinds of fixes, i will try rebooting a few times and doing some stuff inside steam to see if its all okay now, and if it is i will close the issue.

**Comment by VojislavMrdja** [comment]
I'm at my wits end here. Still got no idea what caused the issue or what fixed it, it might be some sort of race condition or something, but steam now opens at least, from the desktop file that is, if I try to do it from the terminal it gets stuck on trying to download some sort of update from an url and just stays there stuck, but at least I can open it from the shortcut. A dev should probably take a look at the logs and the dump I provided to try to figure out what the hell happened/ is happening. Closing as not fixed as I can't reproduce the crashing again.

**Comment by VojislavMrdja** [comment]
Just figured out that pacman apparently had a big update yesterday and went from libalpm.so.14 to libalpm.so.15 which broke yay and some other stuff on my system, so this might just be an arch packaging issue.

**Comment by halozat** [comment]
can confirm, it does not work for me either.
`failed to create dri2 screen`

**Comment by WafiWadud** [comment]
same here

**Comment by rhysperry111** [comment]
Bumping as Steam still crashes for me...

**Comment by tealblu** [comment]
steam also crashing for me :/

**Comment by vrbz** [comment]
im also having this issue, both steam-runtime and since today steam-native has also stopped working, even after full uninstall and removal of files after reinstalling i'm having no luck, hoping somebody finds a fix for this soon

**Comment by theophilusx** [comment]
Hope this may help others. Sorry the info isn't more precise. Got a bad cold and not really too up on things right now.

I've been running steam under both hyprland and i3. All was fine until this week, when I decided it was time for a theme change. After updating my gtk 2, 3 and 4 theme settings and qt 5/6, suddenly I could no longer stqart steam. Startup was stuck in an infinite loop of attempting to start the steamwebhelperui, which would flash up, then crash. I had to kill the steam process to stop it. I tried a full re-install of steam with no luck. I'm not using the steam beta and use the rpm version, not flatpak.

I then created a whole new test account on my system. Logged in and tried to run steam. It worked fine. The test account was running xfce as my fedora started as an xfce spin. I then logged in using i3 which also worke3d and with hyprland and wayland, also working. 

I then went back and reverted allo my theme related changes and everything started to work again. I'm not sure yet exactly which theme chagnes were the issue.  Nothing in the logs was helpful or mentioned anything which looked theme relat3ed - just seemed like a cef issue. 

So my suggestion is, if your running into this issue, try creating a fresh default account running whatever DE you like and leavfe it at defaults wrt themes and see if you can run steam.

**Comment by Fell** [comment]
For me, this issue was either related to **btrfs or slow storage**. I migrated my system from a btrfs multi-device setup to ext4 on RAID0 without changing anything else, and Steam started right up without issues after booting into the new system. I used the exact same SSDs too, so I'm pretty sure the issue is related to btrfs.

I collected some logs from the old setup to help debugging: 
[steam-logs.zip](https://github.com/user-attachments/files/17373510/steam-logs.zip)

**Comment by zelamuss** [comment]
> Hope this may help others. Sorry the info isn't more precise. Got a bad cold and not really too up on things right now.
> 
> I've been running steam under both hyprland and i3. All was fine until this week, when I decided it was time for a theme change. After updating my gtk 2, 3 and 4 theme settings and qt 5/6, suddenly I could no longer stqart steam. Startup was stuck in an infinite loop of attempting to start the steamwebhelperui, which would flash up, then crash. I had to kill the steam process to stop it. I tried a full re-install of steam with no luck. I'm not using the steam beta and use the rpm version, not flatpak.
> 
> I then created a whole new test account on my system. Logged in and tried to run steam. It worked fine. The test account was running xfce as my fedora started as an xfce spin. I then logged in using i3 which also worke3d and with hyprland and wayland, also working.
> 
> I then went back and reverted allo my theme related changes and everything started to work again. I'm not sure yet exactly which theme chagnes were the issue. Nothing in the logs was helpful or mentioned anything which looked theme relat3ed - just seemed like a cef issue.
> 
> So my suggestion is, if your running into this issue, try creating a fresh default account running whatever DE you like and leavfe it at defaults wrt themes and see if you can run steam.

i have the exact same issue now,also hyprland arch and issue appeared after changing my gtk theme

have you found an other solution? i cant login to the new accounts that i created(idk why).

**Comment by rhysperry111** [comment]
I too seemed to be able to get it running as another user perfectly fine...
```
# Main user (1000)
xhost +
chmod 777 -R /run/user/1000/
sudo -iu testuser
# Test user
XDG_RUNTIME_DIR=/run/user/1000 WAYLAND_DISPLAY=wayland-1 steam
```

No closer to narrowing down what's wrong with the environment on my current user though.

**Comment by theophilusx** [comment]
No, I didn't really dig much further once I was able to undo the changes
and things reverted back.

I was playing around with the nwg-* stuff when things broke, such as
nwg-look. Now I'm being a bit smarter and running a qemu/kvm virtual and
doing my playing around in there.

I'm using fedora, so this is not related to arch or any other distro To
setup the alternative account, I just used adduser and ensured the new
user was in the right groups to also have admin privs - for fedora, that
is the 'wheel' group.

Once I reverted all my theme stuff, everything wsorked again in the
original account. In particular, I got the sense that it was the Qt
theming that was the issue, but no evidence to really support that
'feeling'.

For me, I was just playing around with my theming because I was between
projects and a bit bored. Normally, it isn't something I care that much
about, but I do prefer a dark theme. I now have something which is
probably 80% there, but lack the motivation to do the last 20%.
